src/lib/iac/detect-iac.ts

```diff
@@ -84,33 +85,36 @@ async function getProjectTypeForIacFile(filePath: string) {
   return projectType;
 }
 
-async function getDirectoryFiles(root: string) {
+async function getDirectoryFiles(
+  root: string,
+  options: { maxDepth?: number } = {},
+) {
   const iacFiles: IacFileInDirectory[] = [];
   const dirPath = pathLib.resolve(root, '.');
-  const files = glob.sync(
-    pathLib.join(dirPath, '/**/**/*.+(json|yaml|yml|tf)'),
-  );
-
-  for (const fileName of files) {
-    const ext = pathLib.extname(fileName).substr(1);
-    if (Object.keys(projectTypeByFileType).includes(ext)) {
-      const filePath = pathLib.resolve(root, fileName);
-
-      await getProjectTypeForIacFile(filePath)
-        .then((projectType) => {
-          iacFiles.push({
-            filePath,
-            projectType,
-            fileType: ext as IacFileTypes,
-          });
-        })
-        .catch((err: CustomError) => {
-          iacFiles.push({
-            filePath,
-            fileType: ext as IacFileTypes,
-            failureReason: err.userMessage,
-          });
-        });
+  const supportedExtensions = new Set(Object.keys(projectTypeByFileType));
+
+  for (const filePath of walkDirectory(dirPath, {
```

Comment:
minor:
maybe we can rename the `walkDirectory` method to something like:
`yieldNextDirectoryFilepaths` / `getNextDirectoryFilepaths` / whatever ?
as it's slightly unclear in this part that the `walkDirectory` method plans to recursively yield all files paths for each directory level.

Reply:
That's a good point, the MDN docs have a nice example where they use [`makeRangeIterator()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators#Generator_functions), I like it because it highlights the returned value is unusual. Perhaps we should follow the pattern:

```suggestion
const directoryIterator = makeDirectoryIterator(dirpath, { ... });
for (const filePath of directoryIterator) {
```

Thoughts?

Reply:
looks good and much clearer! 💪